tests/func-tests/monitoring_test.go

```diff
@@ -118,6 +119,26 @@ var _ = Describe("[crit:high][vendor:cnv-qe@redhat.com][level:system]Monitoring"
 
 	})
 
+	It("kubevirt_hyperconverged_operator_health_status recording rule should follow firing alerts and kubevirt_hco_system_health_status metric", func() {
+		By("Checking if there are firing alerts with critical or warning impact")
+		alerts, err := promClient.Alerts(context.TODO())
+		Expect(err).ShouldNot(HaveOccurred())
+
+		criticalAlertsAreFiring := areAlertsWithImpactFiring(alerts, "critical")
+		warningAlertsAreFiring := areAlertsWithImpactFiring(alerts, "warning")
+
+		By("Fetching kubevirt_hco_system_health_status and kubevirt_hyperconverged_operator_health_status values")
+		kubevirtHCOSystemHealthStatusValue := getMetricValue(promClient, "kubevirt_hco_system_health_status")
+		kubevirtHyperconvergedOperatorHealthStatusValue := getMetricValue(promClient, "kubevirt_hyperconverged_operator_health_status")
+
+		if criticalAlertsAreFiring || kubevirtHCOSystemHealthStatusValue == "2" {
+			Expect(kubevirtHyperconvergedOperatorHealthStatusValue).To(Equal("2"))
+		} else if warningAlertsAreFiring || kubevirtHCOSystemHealthStatusValue == "1" {
+			Expect(kubevirtHyperconvergedOperatorHealthStatusValue).To(Equal("1"))
+		} else {
+			Expect(kubevirtHyperconvergedOperatorHealthStatusValue).To(Equal("0"))
+		}
+	})
```

Comment:
I think it would be better if you either

1) integrate this test with other alert tests, and verify that, for example, when `KubevirtHyperconvergedClusterOperatorUSModification` is fired, `kubevirt_hyperconverged_operator_health_status` value is 1.

2) maintain the test separately, but you intentionally cause an alert to fire and verify that the metric changes accordingly

Reply:
We considered it, but there are downsides for each of these options. 

Option 1:
At the moment, there are only e2e tests for `warning` impact alerts, as we don't have a `none` alert, and the only `critical` impact alert (`KubevirtHyperconvergedClusterOperatorInstallationNotCompletedAlert`) doesn't have an e2e test. Therefore, the integration will miss these cases.

Option 2:
1. This recording rule depends on many alerts. We can trigger a `warning`alert, and expect the recording rule to equal 1, but other things that we didn't initiate can trigger `critical` alerts and fail the test, although the recording rule functions as expected (i.e. its value is 2).
2. We probably can't trigger all the operator alerts (HCO and its secondary components), so even if we trigger a HCO alert and check the recording rule, this doesn't promise that other alerts will make the recording rule behave the same.

Therefore, @sradco and I agreed that the test will only check an existing state of the cluster, and verify the recording rule value according to it.

Reply:
anyway, I think it is better to use option 1. You can after all the `verifyAlertExists` check the value of `kubevirt_hyperconverged_operator_health_status`.

If it was 0 before and we triggered a warning alert it should be 1.
If it was 1 or 2 before and we triggered a warning alert it should stay the same.
If it was 0 or 1 before and we triggered a critical alert it should be 2.
If it was 2 before and we triggered a critical alert it should stay the same.

Related to the critical, I missed the test for `KubevirtHyperconvergedClusterOperatorInstallationNotCompletedAlert`, but I can add it later, with this new check

Reply:
this could maybe even be done in the `beforeeach`/`aftereach`, you check the alerts before and that the metric is correct, and you check the alerts after and that the metric is updated/correct

Reply:
+1. 
Since we don't have a `none` alert, and your suggestion doesn't include changes in components conditions, I suggest to also keep the new test case as well. It will be the only place we verify that the recording rule behaves as expected considering the conditions.

Reply:
The new test verifies that the recording rule follows `kubevirt_hco_system_health_status` metric, which is based on the conditions. Your suggestion is to  check how the alerts impact the recording rule, but don't check `kubevirt_hco_system_health_status` metric.

Reply:
I will try to integrate it with your suggestion. Thanks!

Reply:
but if you don't provoke any "real" change that triggers the metrics to change, that interdependence between `kubevirt_hco_system_health_status` and `kubevirt_hyperconverged_operator_health_status` is already tested in the Prometheus unit tests

Reply:
updated